backend/src/Designer/Infrastructure/GitRepository/AltinnAppGitRepository.cs

```diff
@@ -547,15 +547,36 @@ public async Task SaveLayoutSettings(string layoutSetName, JsonNode layoutSettin
         /// <param name="layoutSetName">The name of the layout set where the layout belong</param>
         /// <param name="layoutFileName">The name of layout file</param>
         /// <param name="layout">The actual layout that is saved</param>
+        /// <param name="layoutIsInitialForPaymentTask">Boolean value indicating if the layout to add is initial for a payment task</param>
         /// <param name="cancellationToken">An <see cref="CancellationToken"/> that observes if operation is cancelled.</param>
-        public async Task SaveLayout(string layoutSetName, string layoutFileName, JsonNode layout, CancellationToken cancellationToken = default)
+        public async Task SaveLayout(string layoutSetName, string layoutFileName, JsonNode layout, bool layoutIsInitialForPaymentTask = false, CancellationToken cancellationToken = default)
         {
             cancellationToken.ThrowIfCancellationRequested();
             string layoutFilePath = GetPathToLayoutFile(layoutSetName, layoutFileName);
+            if (layoutIsInitialForPaymentTask)
+            {
+                AddPaymentComponentToInitialLayoutForPaymentTask(layout);
```

Comment:
Looks like too much logic in the repository level.
Can you move this logic to the service level and only read/write in repository?